### PR TITLE
AMREX_SPACEDIM : EB, Filter, Laser, Utils, and Diagnostics

### PR DESCRIPTION
**Pull request by prkkumar**
Repalce AMREX_SPACEDIM macros with WARPX_DIM. I believe, this PR completes the task in all the directories!

### CODE REVIEW

File: Source/Laser/LaserProfilesImpl/LaserProfileFromTXYEFile.cpp [hunk 1]
```diff
@@ -159,9 +159,9 @@ WarpXLaserProfiles::FromTXYEFileLaserProfile::parse_txye_file(std::string txye_f
 
         if(m_params.nt <= 1) Abort("nt in txye file must be >=2");
         if(m_params.nx <= 1) Abort("nx in txye file must be >=2");
-#if ((AMREX_SPACEDIM == 3) || (defined WARPX_DIM_RZ))
+#if (defined(WARPX_DIM_3D) || (defined WARPX_DIM_RZ))
         if(m_params.ny <= 1) Abort("ny in txye file must be >=2 in 3D");
-#elif(AMREX_SPACEDIM == 2)
+#elif defined(WARPX_DIM_XZ)
```
Comment (prkkumar) [inline Source/Laser/LaserProfilesImpl/LaserProfileFromTXYEFile.cpp:164]: @lucafedeli88 Is this correct? I usually replace `AMREX_SPACEDIM == 2` with `defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)`. But, here I see 3D and RZ use the same code, so, XZ is the only case left for AMREX_SPACEDIM == 2.

File: Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp [hunk 2]
```diff
@@ -194,10 +194,10 @@ FlushFormatPlotfile::WriteJobInfo(const std::string& dir) const
             jobInfoFile << "   -y: " << "interior" << "\n";
             jobInfoFile << "   +y: " << "interior" << "\n";
         }
-        if (AMREX_SPACEDIM == 3) {
+        #if defined(WARPX_DIM_3D)
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp]:
```suggestion
#if defined(WARPX_DIM_3D)
```

File: Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp [hunk 3]
```diff
@@ -194,10 +194,10 @@ FlushFormatPlotfile::WriteJobInfo(const std::string& dir) const
...
             jobInfoFile << "   -z: " << "interior" << "\n";
             jobInfoFile << "   +z: " << "interior" << "\n";
-        }
+        #endif
```
Comment (ax3l) [inline Source/Diagnostics/FlushFormats/FlushFormatPlotfile.cpp]:
```suggestion
#endif
```

**Review by ax3l — APPROVED** [review]
Thank you so much, LGTM!

Will wait for Luca to have a chance to take a look on your open question.

File: Source/Laser/LaserProfilesImpl/LaserProfileFromTXYEFile.cpp (hunk 1 above)
Comment (ax3l) [reply Source/Laser/LaserProfilesImpl/LaserProfileFromTXYEFile.cpp:164]: I think Luca might be on holidays or time off.

This looks correct to me, so I'll merge it in. Thanks for highlighting!

**Pull request merged**